HPCC-15987 Minor improvements to disk read processing

ghalliday opened the pull request:
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

ghalliday committed:
HPCC-15987 Minor improvements to disk read processing

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday commented:
Jira: https://track.hpccsystems.com/browse/HPCC-15987

ghalliday commented:
@jakesmith FYI
@mckellyln please review

HPCCSmoketest commented:
Automated Smoketest 
Sha: b7a0bdb28d49031f884bc2234e8fd734d4f016cf
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

ghalliday committed:
HPCC-15987 Reorder assignments to optimize return builder.finalizeRowClear()

With the clear after the call to finalize gcc spots that the
destructor can be optimized away.

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

HPCCSmoketest commented:
Automated Smoketest 
Sha: cb4ec4fc0b3305a1d60f39654c891e1f8cfc5c6a
Build: success
Install hpccsystems-platform-community_6.1.0-trunk0.el6.x86_64.rpm
HPCC Start: OK
HPCC Stop: OK
HPCC Uninstall: OK

mckellyln commented:
Changes look fine to me.

richardkchapman commented on rtl/eclrtl/rtlds_imp.hpp:
```diff
@@ -250,17 +250,18 @@ class ECLRTL_API RtlDynamicRowBuilder : implements RtlRowBuilderBase
         maxLength = _maxLength;
     }
-    inline ~RtlDynamicRowBuilder() { clear(); }
+    inline ~RtlDynamicRowBuilder() { if (self) { rowAllocator->releaseRow(self); } }
```
Minor: odd style to have the extra { } here
